### DISCUSSION TITLE
Newbie Question: Handwheel Encoder.

### DISCUSSION THREAD

**Discussion by JaY2119** [kachurovskiy/nanoels#196] [Q&A] [answered]
Hi Maxim/All,

I am currently building H4. I wanted to order from maxim but upon checking everywhere there is no stock yet. So i ended up building the H4. The pcbs arrive last week and i wired everything.

i wanted to add a handwheel encoder for Z and X axis. 
can anyone help me how to wire a handwheel encoder to H4? so i can control Z and X axis.
Also what i need to make it work?

i have a 100ppr 4wire handwheel encoder

thank you in advance guys!!!
Cheers!!

![image](https://github.com/kachurovskiy/nanoels/assets/165907456/556bc824-04b2-4c12-aa93-e22cd92d5ab5)
![image](https://github.com/kachurovskiy/nanoels/assets/165907456/14aa5295-3532-4c19-a69e-97edc500b58e)

**Comment by kachurovskiy** [comment]
Hi @JaY2119 - for each encoder you have a free 4-wire terminal on the back - A1 and A2 - in each terminal left to right pins would be 5V GND A B. You would also need to enable them in the code.

**Reply by JaY2119** [reply]
Hi @kachurovskiy thank you very much for answering my question i really appreciate your answer..
but i forgot to tell earlier that i have zero knowledge on programming IDE.
i scroll down to h4 ino file. but i cant find the way to enable the A1.

is there a chance you can help or show me the row/column on the Arduino IDE that i need to change to enable the A1?

thank you in advance..

**Reply by kachurovskiy** [reply, marked as answer]
No worries, just check the lines 59 and 62 - https://github.com/kachurovskiy/nanoels/blob/ee9b80e4edecbee13295d2b295a7f7feb30526c3/h4/h4.ino#L59 - you'd want to change false to true there and maybe also do the same for lines 61 and 64 to invert the direction of encoders if needed.

**Reply by JaY2119** [reply]
Hi maxim!!
After i got out of the hospital finally i able to test it.
And it works perfectly.
Thanks a lot brother for helping me..

**Comment by Viktor1972-avi** [comment]
Доброе утро. Максим у меня РГИ на 6 выходов подключил к Vcc 5в,  V0 - A11, A - A12 и к B - A-13. В прошивке 59 и 62 строка заменил на true. Перемещений нет. Цепи проверил все хорошо. В прошивке есть ещё настройка которая отвечает за работу с РГИ?

**Reply by kachurovskiy** [reply]
На А12 и А13 есть смена напряжения с 0 на 5 вольт и обратно когда крутите РГИ?

**Reply by Viktor1972-avi** [reply]
Проверю. При  подключении  питания к РГИ на клемах V0 и  5в напряжение падает до 2.9-3 в

**Reply by Viktor1972-avi** [reply]
Проверил между А и В есть изменение напряжения от 0 до 2.9 в при прокручивании РГИ, но оси не двигаются, один раз что то проскочило и все

**Reply by kachurovskiy** [reply]
3 вольта это мало, должно 5 подаваться на РГИ и выходить на сигнальных линиях

**Reply by Viktor1972-avi** [reply]
подается 5, но при подключении рги понижается до 3в

**Reply by kachurovskiy** [reply]
Нужно посмотреть в каком месте находится излишнее сопротивление и происходит падение напряжения

**Comment by Viktor1972-avi** [comment]
Разобрались коллективно, V0 подключили к gnd и все заработало, к А11 ни чего не подключено. Спасибо за помощь.

**Comment by Viktor1972-avi** [comment]
А если на А11 и А21 подключить GND лучше не станет?
